darts/timeseries.py

```diff
@@ -1468,7 +1468,7 @@ def longest_contiguous_slice(self, max_gap_size: int = 0) -> 'TimeSeries':
         max_slice_start = None
         max_slice_end = None
         for index, row in relevant_gaps.iterrows():
-            size = row['gap_start'] - curr_slice_start - self._freq
```

Comment:
Does changing the order of the operands make a difference here?

Reply:
Yes, 
- `row['gap_start'] - curr_slice_start`  gives a time delta in days from which we cannot subtract the frequency.
- `row['gap_start'] - self._freq` gives a time stamp from which we can subtract `curr_slice_start`

Reply:
Could we change our API to avoid relying on the order of the operands? Because this seems a bit fragile... I would expect a subtraction to always return the same type.

Reply:
This is how pandas handles the subtraction
- `pd.Timestamp - pd.Timestamp -> pd.TimeDelta`
- `pd.Timestamp - pandas DateOffset (freq) -> pd.Timestamp`

As TimeDelta doesn't carry information about the exact dates, we cannot subtract **irregular** frequencies (such as month start 'MS', etc.) which is why the original operand order fails. 

For **regular** frequencies (such as 'D') the order doesn't matter for the results.

Maybe we could make it more obvious with:

```
curr_slice_end = row['gap_start'] - self._freq
size = curr_slice_end - curr_slice_start
```
?

Reply:
+1 with your solution (and adding a comment) @dennisbader